crates/bytesbuf/benches/bytesbuf_vs_bytes.rs

```diff
@@ -0,0 +1,1288 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#![expect(missing_docs, reason = "Benchmark code")]
+
+use std::alloc::System;
+use std::hint::black_box;
+use std::num::NonZero;
+
+use alloc_tracker::{Allocator, Session};
+use bytes::{Buf, BufMut};
+use bytesbuf::{BlockSize, BytesBuf, BytesView, FixedBlockTestMemory, TransparentTestMemory};
+use criterion::{BatchSize, Criterion, criterion_group, criterion_main};
+use new_zealand::nz;
+
+criterion_group!(benches, entrypoint);
+criterion_main!(benches);
+
+#[global_allocator]
+static ALLOCATOR: Allocator<System> = Allocator::system();
+
+// The test data is "HTTP request sized". Ultimately, we expect most operations to be zero-copy,
+// so the size of the test data should not matter much, unless we try reading it all at once.
+const TEST_SPAN_SIZE: NonZero<BlockSize> = nz!(12345);
+const TEST_DATA: &[u8] = &[88_u8; TEST_SPAN_SIZE.get() as usize];
+
+const MANY_SPANS: usize = 32;
+const COPY_TO_SLICE_LEN: usize = 256;
+
+#[expect(clippy::too_many_lines, reason = "Is fine - lots of benchmarks to do!")]
+fn entrypoint(c: &mut Criterion) {
+    let allocs = Session::new();
+
+    let memory = FixedBlockTestMemory::new(TEST_SPAN_SIZE);
+    let transparent_memory = TransparentTestMemory::new();
+
+    // Prepare test data - a multi-span view for reading operations
+    let test_data_as_seq = BytesView::copied_from_slice(TEST_DATA, &memory);
+    let many = std::iter::repeat_n(test_data_as_seq.clone(), MANY_SPANS).collect::<Vec<_>>();
+    let many_as_seq = BytesView::from_views(many.iter().cloned());
+
+    // ============================================================================
+    // GET operations (slower, non-numeric)
+    // ============================================================================
+
+    let mut group = c.benchmark_group("get");
```

Comment:
There are quite a lot of benchmarked functions, actually. Let's pair them up, so we have only two entries in each group - one from bytesbuf, one from bytes.

There are some functions that only exist in one:

* bytesbuf has copy_to_slice_uninit() - such we can merge into the "similar group" as just an extra data point, because it is similar to copy_to_slice()
* bytesbuf has put_bytes() which has no direct equivalent in bytes; the bytes::put_bytes() is actually the equivalent of put_slice() in bytesbuf. However, they are all similar enough to be grouped together.

Possibly you will find more that can be grouped thusly.